pro_tes/ga4gh/tes/task_runs.py

```diff
@@ -453,7 +453,16 @@ def _set_projection(self, view: str) -> Dict:
     def _update_task_incoming(
         self, payload: dict, db_document: DbDocument, start_time: str, **kwargs
     ) -> DbDocument:
-        """Update the task incoming object."""
+        """Update the task incoming object.
```

Comment:
Better: "the incoming task document".

However, as the incoming task document is supposed to be immutable, once created (can you check if there is a MongoDB setting that we could use here to enforce this?), you will anyway need to change that in an upcoming PR. Still, better to correct this now so that it doesn't propagate to the next PR.

Reply:
At this stage, the incoming task document will keep updating till the task is completed at the remote TES, by the way, will fix this naming issue in #137. so will it make sense to make it immutable? correct me if I am wrong

Reply:
Fine to address it in #137, but _then_ the incoming task document should indeed be immutable (i.e., the parts that are provided by the client, not the . It's a log/reference to what the user sent, we need this for accounting/audit purposes. We then create the _outgoing_ request and _that_ we can modify according to whatever middleware is configured to act upon the incoming request.

Reply:
Okay, understood but I don't think it is good to fix this in this pr as the naming of the task document is already messed up, we can do this once the naming issue #137 is resolved, if you agree I will create an issue for this.

Reply:
Issue Created #142